### PR TITLE
Added fits extension and the crires wavelength usage

### PR DESCRIPTION
**Pull request by jason-neal**

### CODE REVIEW

File: plot_fits.py [hunk 1]
```diff
@@ -291,12 +291,19 @@ def _parser():
                         'or both.')
     parser.add_argument('--ftype', help='Select which type the fits file is',
                         choices=['ARES', 'CRIRES'], default='ARES')
+    parser.add_argument('--fitsext', help='Select Crires extention',
+                        choices=['0', '1', '2', '3', '4'], default='0')
```
Comment (DanielAndreasen) [inline plot_fits.py]: Why is default 0 and not 1? Also, shouldn't this be integers or is that not possible?

File: plot_fits.py [hunk 2]
```diff
@@ -291,12 +291,19 @@ def _parser():
...
 
 
 def main(fname, lines=False, model=False, telluric=False, sun=False,
-         rv=False, rv1=False, rv2=False, ccf='none', ftype='ARES'):
+         rv=False, rv1=False, rv2=False, ccf='none', ftype='ARES', 
+<<<<<<< HEAD
+         fitsext='0'):
+=======
+         fitsext='none'):
+>>>>>>> fe8f32765887481754cd38bd686c1e7e24deb0ab
```
Comment (DanielAndreasen) [inline plot_fits.py]: It looks like you have a merge conflict here. Please select which option you prefer.

File: plot_fits.py [hunk 3]
```diff
@@ -334,21 +342,28 @@ def main(fname, lines=False, model=False, telluric=False, sun=False,
         _download_spec(pathsun)
         print('Downloading telluric spectrum...')
         _download_spec(pathtel)
+    
+    fitsext = int(fitsext)
+<<<<<<< HEAD
```
Comment (DanielAndreasen) [inline plot_fits.py]: Another merge conflict

File: plot_fits.py [hunk 4]
```diff
@@ -334,21 +342,28 @@ def main(fname, lines=False, model=False, telluric=False, sun=False,
         _download_spec(pathsun)
         print('Downloading telluric spectrum...')
         _download_spec(pathtel)
+    
+    fitsext = int(fitsext)
```
Comment (DanielAndreasen) [inline plot_fits.py]: Could be nice if this is set directly from the `argparse` function.

**Comment by DanielAndreasen** [comment]
Hi Jason,

Thank you very much for this pull request. I made some comments in the code. If you can clean this up a little, then I can merge it.  
Another thing that could be nice, but I don't know how that would work is, that the new dropdown menu only appeared if one choose CRIRES.

Cheers,
Daniel

File: plot_fits.py (hunk 1 above)
Comment (jason-neal) [inline plot_fits.py]: The default of 0 returns primary header, and works when there are no extensions. I have also added the fitsext to the getdata call for ARES but with the default of 0 it should still work.  Returns same data as extension 1 with the CRIRES data. Integers are possible with trade offs, see below.

**Comment by jason-neal** [comment]
Hi Daniel,

I fixed the merge conflicts. But as for the other comments I have left them pretty much them the same.
I experimented with other widget types but returned to the string solution for the ease of the drop down menu. I have left one solution commented out, which does define an integer input but it requires you to type in the extension number you want.   
I tried a list of integers but Gooey throws a TypeError: must be string or Unicode.
I also tried the 'count' widget but that gives a dropdown list of integers 1-10 a bit more than necessary.

I am not sure about how to implement the dis/appearing dropdown. Possibly would need to involve the mutually exclusive functionality. 
Possibly instead of dropdown lists have a tickboxes.
[] ARES     [] CRIRES ext 1
[] CRIRES ext 2

File: plot_fits.py [hunk 5]
```diff
@@ -309,6 +314,7 @@ def main(fname, lines=False, model=False, telluric=False, sun=False,
     :rv2: RV of telluric spectrum
     :ccf: Calculate CCF (sun, model, telluric, both)
     :ftype: Type of fits file (ARES or CRIRES)
+    :fitsext: Slecet fits extention to use (0,1,2,3,4)
```
Comment (jason-neal) [inline plot_fits.py]: Sorry for the spelling...

**Pull request merged**